vmdb/lib/token_manager.rb

```diff
@@ -36,94 +36,42 @@ def self.configure(_namespace, options = {})
   end
 
   def self.gen_token(namespace, token_options = {})
-    token_cleanup
-    token = SecureRandom.hex(16)
-
     ts = global_token_store(namespace)
-    ts[token] = {:expires_on => Time.now.tv_sec + @config[:token_ttl]}
-    ts[token].merge!(prune_token_options(namespace, token, token_options))
+    token = SecureRandom.hex(16)
+    token_data = {:expires_on => Time.now.tv_sec + @config[:token_ttl]}
 
+    ts.add(token,
+           token_data.merge!(prune_token_options(token_options)),
+           @config[:token_ttl])
     token
   end
 
   def self.token_set_info(namespace, token, token_options = {})
     ts = global_token_store(namespace)
-    return {} unless ts.key?(token)
-    ts[token].merge!(prune_token_options(namespace, token, token_options))
+    token_data = ts.get(token)
+    return {} if token_data.blank?
+
+    ts.set(token, token_data.merge!(prune_token_options(token_options)))
   end
 
   def self.token_get_info(namespace, token, what = nil)
     ts = global_token_store(namespace)
-    return {} unless ts.key?(token)
-    what.nil? ? ts[token] : ts[token][what]
-  end
+    return {} unless token_valid?(namespace, token)
 
-  def self.token_valid?(namespace, token)
-    global_token_store(namespace).key?(token)
+    what.nil? ? ts.get(token) : ts.get(token)[what]
   end
 
-  def self.token_expired?(namespace, token)
-    return true unless token_valid?(namespace, token)
-
-    ts = global_token_store(namespace)
-    if ts[token][:expires_on] < Time.now.tv_sec
-      log "#{namespace}:Expiring Token #{token}"
-      ts.delete(token)
-      return true
-    end
-    false
+  def self.token_valid?(namespace, token)
+    !global_token_store(namespace).get(token).nil?
```

Comment:
I didn't want the token_valid? method to return token details, just ye-old true/false.